Atanu0341/countries
# feat: add empty state for CountryList and improve error handling

### Summary

This PR introduces an improved UX for the CountryList component. When no countries match the search query, a user-friendly message is shown.

fix: #1 

### Changes Implemented:
- Added an empty state with a message when no countries match the search term.

### Testing Instructions:
1. Run the app locally.
2. Search for a country that does not exist.
3. Ensure that a friendly message is displayed.
4. Ensure the app is fully responsive.

### Screenshots :
![Screenshot 2024-10-17 152848](https://github.com/user-attachments/assets/4e69a845-c196-4e95-8b0e-8c0b47b5dd98)


### Related Issue:
Closes #1
